pkg/bootstrapper/bootstrapper.go

```diff
@@ -830,3 +828,58 @@ func (cni *cniOptions) configure(kubeletCmd *string) error {
 
 	return nil
 }
+
+// convertKubeletArgsToMap converts the list of kubelet arguements into a map. This is required
+// to reconstruct kubelet command when updating the kubelet service
+func convertKubeletArgsToMap(kubeletArgs []string) (map[string]string, error) {
+	kubeletKeyValueArgs := make(map[string]string)
+
+	for _, option := range kubeletArgs {
+		// Args like --config=c:\\k\\kubelet.conf will be split on "=" and stored as key value pairs of the map.
+		//Stand alone args like --windows-service will be stored as a string against a special key
+		if strings.Contains(option, "=") {
+			kv := strings.Split(option, "=")
+			kubeletKeyValueArgs[kv[0]] = kv[1]
+			// This is to account for args like --register-with-taints=os=Windows:NoSchedule
+			if len(kv) > 2 {
+				for _, val := range kv[2:] {
+					kubeletKeyValueArgs[kv[0]] += "=" + val
+				}
+			}
+		} else {
+			kubeletKeyValueArgs[kubeletStandAloneArgsKey] += option + " "
+		}
+	}
+
+	// Remove the trailing space
+	if standaloneArgs, found := kubeletKeyValueArgs[kubeletStandAloneArgsKey]; found {
+		kubeletKeyValueArgs[kubeletStandAloneArgsKey] = strings.TrimSpace(standaloneArgs)
+	}
+	return kubeletKeyValueArgs, nil
+
+}
+
+// configureKubelet updates the given kubelet command with the required args.
```

Comment:
If this is updating the kubelet cmd then why is it called `configureKubelet`? This should be `updateKubeletCmd`